### PR TITLE
[SYCL] Add device copyable trait for annotated_arg

### PR DESCRIPTION
**Pull request by yug-intel**
Added device copyable traits for annotated_arg. If the type `T` is device copyable, then the type `annotated_arg<T, ...>` is also device copyable.

For example, consider the following code: 

```c++
#include <sycl/sycl.hpp>
#include <sycl/ext/intel/fpga_extensions.hpp>
#include <sycl/ext/intel/ac_types/ac_fixed.hpp>
#include "ExceptionHandler.h"
#include "TestConfigSelector.h"

using namespace sycl;
using namespace ext::intel::experimental;
using namespace ext::oneapi::experimental;

using ac_fixed_type = ac_fixed<10, 5, true>;

int main() {
    struct MyIP {
        annotated_arg<ac_fixed_type> a;
        void operator()() const {};
    };
    queue q(testconfig_selector_v, &m_exception_handler);
    ac_fixed_type a = 5;
    q.single_task(MyIP{a}).wait();
    return 0;
}
```

This code generates the following static assertion failure without the device copyable trait for `annotated_arg`: 

```
/nfs/site/disks/swbld_rel_hld/r/sycl/trunk/20230620/linux64/bin-llvm/../include/sycl/types.hpp:2385:17: error: static assertion failed due to requirement 'is_device_copyable<sycl::ext::oneapi::experimental::annotated_arg<ac_fixed<10, 5, true, AC_TRN, AC_WRAP>, sycl::ext::oneapi::experimental::properties<std::tuple<>>>, void>::value || detail::IsDeprecatedDeviceCopyable<sycl::ext::oneapi::experimental::annotated_arg<ac_fixed<10, 5, true, AC_TRN, AC_WRAP>, sycl::ext::oneapi::experimental::properties<std::tuple<>>>, void>::value': The specified type is not device copyable                        
 2385 |   static_assert(is_device_copyable<FieldT>::value ||
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2386 |                     detail::IsDeprecatedDeviceCopyable<FieldT>::value,
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### CODE REVIEW

**Review by bso-intel — APPROVED** [review]

**Comment by bader** [comment]
@yug-intel, please, follow contribution guidelines for PR title and description decoration - https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md#pull-request.

**Comment by tiwaria1** [comment]
Yug can you check with `annotated_ptr<ac_fixed, ...>` if this is even needed for `annotated_ptr`? I don't think it is needed because `annotated_ptr` is a simple pointer wrapper and hence must be trivially copyable (haven't checked). If so, do not add the traits for `annotated_ptr`.

**Review by GarveyJoe — APPROVED** [review]

**Review by tiwaria1 — APPROVED** [review]

**Comment by yug-intel** [comment]
@bso-intel Could you review the spec doc change and merge PR if it looks good?

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_arg.asciidoc [hunk 1]
```diff
@@ -179,6 +179,19 @@ The properties supported with `annotated_arg` may be defined in
...
+as defined by the following:
+
+[source,c++]
+----
+template <typename T, typename PropertyList>
+struct is_device_copyable<
+    ext::oneapi::experimental::annotated_arg<T, PropertyList>,
+    std::enable_if_t<!std::is_trivially_copyable_v<
+        ext::oneapi::experimental::annotated_arg<T, PropertyList>>>>
+    : is_device_copyable<T> {};
```
Comment (tiwaria1) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_arg.asciidoc:192]: This is more like implementation detail, I don't think we need this trait here. Your description is sufficient.

**Review by tiwaria1 — APPROVED** [review]

**Review by bso-intel — APPROVED** [review]

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_arg.asciidoc [hunk 2]
```diff
@@ -179,6 +179,8 @@ The properties supported with `annotated_arg` may be defined in
 separate extensions. Please note that there cannot be duplicated property in a 
 properties list. Otherwise, a compiler time error is triggered.
 
+The type `annotated_arg<T, ...>` is device copyable given that the type `T` is device copyable.
```
Comment (gmlueck) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_arg.asciidoc:182]: We should mention trivially copyable.  How about:

> If the type `T` is trivially copyable, then `annotated_arg<T, ...>` is also trivially copyable.
>
> If the type `T` is device copyable, then `annotated_arg<T, ...>` is also device copyable and the implementation sets the `is_device_copyable_v` trait to `true` for this type.

**Review by gmlueck — APPROVED** [review]

**Pull request merged**